Generator/Extensions/TypeExtensions.cs

```diff
@@ -0,0 +1,7 @@
+﻿namespace Generator.Extensions;
+
+public static class TypeExtensions
+{
+    public static bool IsConcreteTypeOfSomethingPolymorphic(this Type type) => ((type.BaseType != typeof(object) && type.BaseType is not null) || type.GetInterfaces().Length > 0) && !type.IsAbstract;
```

Comment:
```suggestion
    public static bool IsConcreteTypeOfSomethingPolymorphic(this Type type) => ((type.BaseType is not null && type.BaseType != typeof(object)) || type.GetInterfaces().Length > 0) && !type.IsAbstract;
```

Very small optimization, but we can do the null check on type.BaseType before comparing its type.